rg-cleanup.sh

```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
+
+set -euo pipefail
+
+# Parse out --role-assignments flag
+role_assignments_flag=false
+while [[ "$#" -gt 0 ]]; do
+  case $1 in
+    --role-assignments) role_assignments_flag=true ;;
+    *) args+=("$1") ;;  # Store other arguments in an array
+  esac
+  shift
+done
+
+# Call resource group cleanup binary with its command-line arguments
+./bin/rg-cleanup "${args[@]}"
+
+# Exit if not cleaning up unattached role assignments
+if [ "$role_assignments_flag" != true ]; then
+  echo "Skipping unattached role assignment cleanup..."
+  exit 0
+fi
+
+# Clean up unattached role assignments
+az login --identity
+az account set --subscription "${SUBSCRIPTION_ID}"
+
+echo "Cleaning up unattached role assignments..."
+QUERY=".[] | select(.scope == \"/subscriptions/${SUBSCRIPTION_ID}\" and .principalName==\"\") .id"
+for R_ID in $(az role assignment list --all --include-inherited -o json | jq -r "${QUERY}"); do
+  echo "Deleting unattached role assignment: $R_ID"
+  az role assignment delete --ids $R_ID
+done
```

Comment:
I think this should be equivalent and might have a better shot at reporting errors that could be swallowed somewhere:

```suggestion
assignments=$(az role assignment list --scope "/subscriptions/$SUBSCRIPTION_ID" -o tsv --query "[?principalName==''].id")
echo "Deleting role assignments:"
echo "$assignments"
xargs az role assignment delete --ids <<< "$assignments"
```

Reply:
Let's get this debug foo in and merge this PR and start the stdout/stderr carousel. :)

Reply:
I'll try deploying this version to the rg-cleanup logic app later this morning, cross fingers. I'm running the manual cleanup right now as a workaround.